src/track/beatgrid.cpp

```diff
@@ -369,3 +388,34 @@ void BeatGrid::setBpm(double dBpm) {
     locker.unlock();
     emit updated();
 }
+
+void BeatGrid::setSignature(mixxx::TimeSignature sig, double dSample) {
+    Q_UNUSED(dSample);
+    QMutexLocker locker(&m_mutex);
+    if (!isValid())
+        return;
+
+    m_grid.mutable_first_beat()->mutable_signature()->set_beats(sig.getBeats());
+    m_grid.mutable_first_beat()->mutable_signature()->set_note_value(sig.getNoteValue());
+}
+
+mixxx::TimeSignature BeatGrid::getSignature(double dSample) const {
+    Q_UNUSED(dSample);
+    QMutexLocker locker(&m_mutex);
+    if (!isValid())
+        return mixxx::TimeSignature(0,0);
+
+    // If no signature assume 4/4
+    if (!m_grid.first_beat().has_signature()) {
+        return mixxx::TimeSignature(4,4);
+    } else {
+        return mixxx::TimeSignature(
+                m_grid.first_beat().signature().beats(),
+                m_grid.first_beat().signature().note_value());
+    }
+}
+
+void BeatGrid::setBar(double dSample) {
```

Comment:
setDownBeat or similar

Reply:
BeatGrid is going to disappear so I will apply the change on the refurbished Beats class.

Reply:
Done in new Beats